src/pages/images/ImageList.tsx

```diff
@@ -42,6 +46,16 @@ const ImageList: FC = () => {
     notify.failure("Loading images failed", error);
   }
 
+  useEffect(() => {
+    const validNames = new Set(images?.map((image) => image.fingerprint));
+    const validSelections = selectedNames.filter((name) =>
```

Comment:
this could be slightly more efficient, we could exit early if we found a mismatch?

```
for (const name of selectedNames) {
  if (!validNames.has(name)) {
    setSelectedNames(validSelections);
    break;
  }
}
```

Reply:
We have to calculate the `validSelections` as the overlap between selections and valid names. I don't see a way to do that without iterating and checking all selected items for validity.

Reply:
aah yes you are right, please disregard.